src/main/kotlin/no/nav/syfo/client/behandlendeenhet/BehandlendeEnhetClient.kt

```diff
@@ -26,6 +29,31 @@ class BehandlendeEnhetClient(
         callId: String,
         personident: Personident,
         token: Token,
+    ): BehandlendeEnhetDTO {
+        val cacheKey = "$BEHANDLENDEENHET_CACHE_KEY-$personident"
+        val cachedEnhet = getCachedBehandlendeEnhet(cacheKey)
+
+        return if (cachedEnhet != null) {
+            cachedEnhet
+        } else {
+            val behandlendeEnhet = getBehandlendeEnhet(callId, personident, token)
+            redisStore.setObject(
+                key = cacheKey,
+                value = behandlendeEnhet,
+                expireSeconds = PdlClient.TWELVE_HOURS_IN_SECS
+            )
+            behandlendeEnhet
+        }
+    }
+
+    private suspend fun getCachedBehandlendeEnhet(cacheKey: String): BehandlendeEnhetDTO? {
+        return redisStore.getObject(key = cacheKey)
+    }
+
+    private suspend fun getBehandlendeEnhet(
```

Comment:
Jeg er litt usikker på dette navnet. Den burde kanskje hete noe som `getEnhetFromSyfobehandlendeenhet`, for å skille den fra henting fra cache, men så ser det litt rart ut? 🤔

Reply:
Den er jo private så spiller kanskje ikke så stor rolle. Men getEnhetFromSyfobehandlendeenhet funker bra.